Source/WebCore/rendering/RenderBox.cpp

```diff
@@ -5491,9 +5493,19 @@ LayoutBoxExtent RenderBox::scrollPaddingForViewportRect(const LayoutRect& viewpo
         minimumValueForLength(padding.bottom(), viewportRect.height()), minimumValueForLength(padding.left(), viewportRect.width()));
 }
 
-LayoutUnit synthesizedBaselineFromBorderBox(const RenderBox& box, LineDirectionMode direction)
+LayoutUnit synthesizedBaseline(const RenderBox& box, const RenderStyle* parentStyle, LineDirectionMode direction, BaselineSynthesisEdge edge)
 {
-    return direction == HorizontalLine ? box.height() : box.width();
+    ASSERT(parentStyle);
+    LayoutUnit boxSize = direction == HorizontalLine ? box.height() : box.width();
+
+    if (edge == Content)
+        boxSize -= direction == HorizontalLine ? box.verticalBorderAndPaddingExtent() : box.horizontalBorderAndPaddingExtent();
+    if (edge == Margin)
```

Comment:
please use else if (as now it looks like edge is an option set)

Reply:
Changed